casr/src/bin/casr-js.rs

```diff
@@ -0,0 +1,166 @@
+                .value_name("FILE")
+                .help("Stdin file for program"),
+        )
+        .arg(
+            Arg::new("timeout")
+                .short('t')
+                .long("timeout")
+                .action(ArgAction::Set)
+                .default_value("0")
+                .value_name("SECONDS")
+                .help("Timeout (in seconds) for target execution, 0 value means that timeout is disabled")
+                .value_parser(clap::value_parser!(u64).range(0..))
+        )
+        .arg(
+            Arg::new("ignore")
+                .long("ignore")
+                .action(ArgAction::Set)
+                .value_parser(clap::value_parser!(PathBuf))
+                .value_name("FILE")
+                .help("File with regular expressions for functions and file paths that should be ignored"),
+        )
+        .arg(
+            Arg::new("ARGS")
+                .action(ArgAction::Set)
+                .num_args(1..)
+                .last(true)
+                .help("Add \"-- <path> <arguments>\" to run"),
+        )
+        .get_matches();
+
+    init_ignored_frames!("js"); //TODO
+    if let Some(path) = matches.get_one::<PathBuf>("ignore") {
+        util::add_custom_ignored_frames(path)?;
+    }
+    // Get program args.
+    let argv: Vec<&str> = if let Some(argvs) = matches.get_many::<String>("ARGS") {
+        argvs.map(|s| s.as_str()).collect()
+    } else {
+        bail!("Wrong arguments for starting program");
+    };
+
+    // Get stdin for target program.
+    let stdin_file = util::stdin_from_matches(&matches)?;
+
+    // Get timeout
+    let timeout = *matches.get_one::<u64>("timeout").unwrap();
+
+    // Run program.
+    let mut js_cmd = Command::new(argv[0]);
+    if let Some(ref file) = stdin_file {
+        js_cmd.stdin(std::fs::File::open(file)?);
+    }
+    if argv.len() > 1 {
+        js_cmd.args(&argv[1..]);
+    }
+    let js_result = util::get_output(&mut js_cmd, timeout, true)?;
+
+    let js_stderr = String::from_utf8_lossy(&js_result.stderr);
+
+    // Create report.
+    let mut report = CrashReport::new();
+    // Set executable path.
+    report.executable_path = argv[0].to_string();
+    let mut path_to_tool = PathBuf::new();
+    path_to_tool.push(argv[0]);
+    if argv.len() > 1 {
+        if let Some(fname) = Path::new(argv[0]).file_name() {
+            let Ok(full_path_to_tool) = which::which(fname) else {
+                bail!("Could not get the full path of {}", argv[0]);
+            };
+            path_to_tool = full_path_to_tool;
+            let fname = fname.to_string_lossy();
+            if (fname.ends_with("node") || fname.ends_with("jsfuzz"))
+                && !fname.ends_with(".js")
+                && argv[1].ends_with(".js")
+            {
+                report.executable_path = argv[1].to_string();
+            } else if argv.len() > 2
+                && fname.ends_with("npx")
+                && !fname.ends_with(".js")
```

Comment:
it seems unnecessary due to the condition above

Reply:
Done